app/models/game.rb

```diff
@@ -47,10 +47,23 @@ def create_piece(x, y, piece)
 
   def check?(white)
     king = pieces.find_by(type: 'King', white: white)
-    other_pieces = pieces.where(" white != ? and x_coordinate > ?", white, 0)
+    other_pieces = pieces.where("white != ? and x_coordinate > ?", white, 0)
     other_pieces.each do |piece|
       return true if piece.is_valid?(king.x_coordinate, king.y_coordinate)
     end
     false
   end
+
+  def stalemate?(white)
+    turn_pieces = pieces.where(white: white, taken: false)
+    pieces.each do |piece|
```

Comment:
should it be `turn_pieces`? @eyoshimoriyama

Reply:
Hmm yeah I guess that name is not exactly clear. Did you have something in mind? Perhaps current_pieces ?

Reply:
if I read it correctly would a good name be current_player_pieces (I think I have this round the right way ;)).
Returning false if move does not lad to stalemate looks okay to me, works with what I built out for Pawn Promotion too.